config/size-report.js

```diff
@@ -0,0 +1,217 @@
+const path = require('path');
+const fs = require('fs');
+const util = require('util');
+
+const gzipSize = require('gzip-size');
+const fetch = require('node-fetch');
+const prettyBytes = require('pretty-bytes');
+const escapeRE = require('escape-string-regexp');
+const chalk = new require('chalk').constructor({ level: 4 });
+
+const readdir = util.promisify(fs.readdir);
+const stat = util.promisify(fs.stat);
+
+function getTravis(path) {
+  return fetch('https://api.travis-ci.org' + path, {
+    headers: { 'Travis-API-Version': '3' },
+  });
+}
+
+function getTravisJson(path) {
+  return getTravis(path).then(r => r.json());
+}
+
+function getTravisText(path) {
+  return getTravis(path).then(r => r.text());
+}
+
+/**
+ * Recursively-read a directory and turn it into an array of { name, size, gzipSize }
+ */
+async function dirToInfoArray(startPath, {
+  namePrefix = '',
+} = {}) {
+  const entries = await readdir(startPath, { withFileTypes: true });
+  const promises = [];
+  const result = [];
+
+  for (const entry of entries) {
+    const entryPath = path.join(startPath, entry.name);
+    if (entry.isFile()) {
+      promises.push(async function() {
+        const gzipPromise = gzipSize.file(entryPath);
+        const statPromise = stat(entryPath);
+
+        result.push({
+          name: namePrefix + entry.name,
+          gzipSize: await gzipPromise,
+          size: (await statPromise).size,
+        });
+      }());
+    } else if (entry.isDirectory()) {
+      promises.push(async function() {
+        const dirResult = await dirToInfoArray(entryPath, {
+          namePrefix: entry.name + '/',
+        });
+
+        result.push(...dirResult);
+      }());
+    }
+  }
+
+  await Promise.all(promises);
+  return result;
+}
+
+/**
+ * Try to treat two entries with different file name hashes as the same file.
+ */
+function findHashedMatch(name, buildInfo) {
+  const nameParts = /^(.+\.)[a-f0-9]+(\..+)$/.exec(name);
+  if (!nameParts) return;
+
+  const matchRe = new RegExp(`^${escapeRE(nameParts[1])}[a-f0-9]+${escapeRE(nameParts[2])}$`);
+  const matchingEntry = buildInfo.find(entry => matchRe.test(entry.name));
+  return matchingEntry;
+}
+
+const buildSizePrefix = '=== BUILD SIZES: ';
+const buildSizePrefixRe = new RegExp(`^${escapeRE(buildSizePrefix)}(.+)$`, 'm');
+
+async function getPreviousBuildInfo() {
+  const buildData = await getTravisJson('/repo/GoogleChromeLabs%2Fsquoosh/builds?branch.name=size-report&state=passed&limit=1');
```

Comment:
I’m guessing switching this to `master` is that “small change”? 😄

Reply:
Yeeep 😄